pkg/neighborsync/neighborsync.go

```diff
@@ -0,0 +1,294 @@
+	neighborsMu.Lock()
+	delete(neighbors, key)
+	neighborsMu.Unlock()
+}
+
+func sendNeighborRequest(linkIndex int, destination net.HardwareAddr, address netip.Addr) {
+	switch {
+	case address.Is4():
+		sendARPRequest(linkIndex, destination, address)
+	case address.Is6():
+		sendNDPRequest(linkIndex, destination, address)
+	default:
+		ctrl.Log.Error(fmt.Errorf("unsupported IP address type: %s", address), "sendNeighborRequest failed")
+	}
+}
+
+func sendARPRequest(linkIndex int, destination net.HardwareAddr, address netip.Addr) {
+	iface, err := net.InterfaceByIndex(linkIndex)
+	if err != nil {
+		return
+	}
+	c, err := arp.Dial(iface)
+	if err != nil {
+		return
+	}
+	defer c.Close()
+
+	ip, err := getFirstIPv4FromInterface(iface)
+	if err != nil {
+		ctrl.Log.Error(err, "failed to get IPv4 address from interface", "interface", iface.Name)
+		return
+	}
+
+	arpPacket, err := arp.NewPacket(arp.OperationRequest, iface.HardwareAddr, ip, destination, address)
+	if err != nil {
+		ctrl.Log.Error(err, "failed to create ARP packet", "interface", iface.Name, "address", address)
+		return
+	}
+
+	if err := c.WriteTo(arpPacket, destination); err != nil {
+		ctrl.Log.Error(err, "sendARPRequest failed", "interface", iface.Name, "address", address)
+	}
+}
+
+func sendNDPRequest(linkIndex int, destination net.HardwareAddr, address netip.Addr) {
+	iface, err := net.InterfaceByIndex(linkIndex)
+	if err != nil {
+		return
+	}
+
+	c, _, err := ndp.Listen(iface, ndp.LinkLocal)
+	if err != nil {
+		ctrl.Log.Error(err, "failed to listen for NDP messages", "interface", iface.Name)
+		return
+	}
+	defer c.Close()
+
+	m := &ndp.NeighborSolicitation{
+		TargetAddress: address,
+		Options: []ndp.Option{
+			&ndp.LinkLayerAddress{
+				Direction: ndp.Source,
+				Addr:      iface.HardwareAddr,
+			},
+			&ndp.LinkLayerAddress{
+				Direction: ndp.Target,
+				Addr:      destination,
+			},
+		},
+	}
+
+	if err := c.WriteTo(m, nil, address); err != nil {
+		ctrl.Log.Error(err, "sendNDPRequest failed", "address", address)
+	}
+}
+
+func getFirstIPv4FromInterface(iface *net.Interface) (netip.Addr, error) {
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return netip.Addr{}, fmt.Errorf("failed to get addresses for interface %s: %w", iface.Name, err)
+	}
+	for _, addr := range addrs {
+		ipNet, ok := addr.(*net.IPNet)
+		if !ok {
+			continue
+		}
+		ipBytes := ipNet.IP.To4()
+		if ipBytes != nil {
+			ip, ok := netip.AddrFromSlice(ipBytes)
+			if ok && ip.Is4() {
+				return ip, nil
+			}
+		}
+	}
+	return netip.Addr{}, fmt.Errorf("no valid IPv4 address found on interface %s", iface.Name)
+}
+
+func processUpdate(update *netlink.NeighUpdate) {
+	logger := ctrl.Log.WithName("neighborsync")
+
+	intf, err := net.InterfaceByIndex(update.Neigh.LinkIndex)
+	if err != nil {
+		return
+	}
+
+	if !shouldTrackInterface(intf.Name) {
+		return
+	}
+
+	addr, ok := netip.AddrFromSlice(update.Neigh.IP)
+	if !ok {
+		return
+	}
+
+	switch update.Type {
+	case unix.RTM_NEWNEIGH:
```

Comment:
Do we have more `update.Type`s we should check for?

Reply:
https://www.man7.org/linux/man-pages/man7/rtnetlink.7.html